OptimizelySDK/Optimizely.cs

```diff
@@ -188,9 +260,16 @@ public static String SDK_TYPE
             EventBuilder = new EventBuilder(Bucketer, Logger);
             UserProfileService = userProfileService;
             NotificationCenter = notificationCenter ?? new NotificationCenter(Logger);
-            DecisionService = new DecisionService(Bucketer, ErrorHandler, userProfileService, Logger);
-            EventProcessor = eventProcessor ?? new ForwardingEventProcessor(EventDispatcher, NotificationCenter, Logger);
-            DefaultDecideOptions = defaultDecideOptions ?? new OptimizelyDecideOption[] { };
+            DecisionService =
+                new DecisionService(Bucketer, ErrorHandler, userProfileService, Logger);
+            EventProcessor = eventProcessor ?? new ForwardingEventProcessor(EventDispatcher,
+                NotificationCenter,
+                Logger);
+            DefaultDecideOptions = defaultDecideOptions ?? new OptimizelyDecideOption[]
+                { };
+#if USE_ODP
+            OdpManager = odpManager ?? new OdpManager.Builder().Build();
```

Comment:
Maybe we need a NoOpOdpManager?

Reply:
I think it's best to pass NoOpOdpManager if you wish to disable ODP. Otherwise, I think it's best to init with at least a default, functional OdpManager which is wired and ready if settings become available.

A couple of unit tests started failing despite having ODP settings in the mock datafile. 

Thoughts @mnoman09 @jaeopt ?

Reply:
I think the best approach would be to use NoOpODPManager but in java we are just not initializing it and I think that is fine too. @jaeopt can suggest better.

Reply:
That's a good idea.  Just leave it `default`. We could then remove the new NoOp. 

I'll hold for @jaeopt 's opinion.

Reply:
Either way looks good to me too.